Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/JointLimitRotationManipulators.cpp

```diff
@@ -0,0 +1,162 @@
+    }
+
+    void JointLimitRotationManipulators::ResetValues()
+    {
+        if (m_physicsSetupManipulatorData.HasJointLimit())
+        {
+            BeginEditing();
+            GetLocalOrientation() = AZ::Quaternion::CreateIdentity();
+            FinishEditing();
+            Refresh();
+        }
+    }
+
+    void JointLimitRotationManipulators::OnManipulatorMoved(const AZ::Quaternion& rotation)
+    {
+        m_rotationManipulators.SetLocalOrientation(rotation);
+        if (m_physicsSetupManipulatorData.HasJointLimit())
+        {
+            GetLocalOrientation() = rotation;
+        }
+        m_physicsSetupManipulatorData.m_jointLimitWidget->InvalidateValues();
+    }
+
+    void JointLimitRotationManipulators::BeginEditing()
+    {
+        if (!m_commandGroup.IsEmpty())
+        {
+            return;
+        }
+        m_commandGroup.SetGroupName("Adjust joint limit");
+        const AZ::u32 actorId = m_physicsSetupManipulatorData.m_actor->GetID();
+        const AZStd::string& nodeName = m_physicsSetupManipulatorData.m_node->GetNameString();
+        CommandAdjustJointLimit* command = aznew CommandAdjustJointLimit(actorId, nodeName);
+        m_commandGroup.AddCommand(command);
+        command->SetOldJointConfiguration(m_physicsSetupManipulatorData.m_jointConfiguration);
+    }
+
+    void JointLimitRotationManipulators::FinishEditing()
+    {
+        if (m_commandGroup.IsEmpty())
+        {
+            return;
+        }
+
+        if (CommandAdjustJointLimit* command = azdynamic_cast<CommandAdjustJointLimit*>(m_commandGroup.GetCommand(0)))
+        {
+            command->SetJointConfiguration(m_physicsSetupManipulatorData.m_jointConfiguration);
+        }
+        AZStd::string result;
+        CommandSystem::GetCommandManager()->ExecuteCommandGroup(m_commandGroup, result);
+        m_commandGroup.Clear();
+    }
+
+    void JointLimitRotationManipulators::OnTick([[maybe_unused]] float delta, [[maybe_unused]] AZ::ScriptTimePoint timePoint)
+    {
+        const AzFramework::CameraState cameraState = AzToolsFramework::GetCameraState(m_viewportId);
+        m_rotationManipulators.RefreshView(cameraState.m_position);
+    }
+
+    void JointLimitRotationManipulators::OnUnderlyingPropertiesChanged()
+    {
+        Refresh();
+    }
+
+    AZ::Quaternion& JointLimitRotationManipulators::GetLocalOrientation()
+    {
+        return m_jointLimitFrame == JointLimitFrame::Parent
+            ? m_physicsSetupManipulatorData.m_jointConfiguration->m_parentLocalRotation
+            : m_physicsSetupManipulatorData.m_jointConfiguration->m_childLocalRotation;
+    }
+
+    const AZ::Quaternion& JointLimitRotationManipulators::GetLocalOrientation() const
+    {
+        return m_jointLimitFrame == JointLimitFrame::Parent
+            ? m_physicsSetupManipulatorData.m_jointConfiguration->m_parentLocalRotation
+            : m_physicsSetupManipulatorData.m_jointConfiguration->m_childLocalRotation;
```

Comment:
Just out of curiousity, who is updating these rotations at runtime or are these from the bind pose?

Reply:
They can also be adjusted in the reflected property editor.